components/speedreader/rust/lib/src/readability/src/extractor.rs

```diff
@@ -203,6 +416,137 @@ pub fn clean_title(dom: &Sink, title: String) -> String {
     }
 }
 
+#[derive(Error, Debug, PartialEq)]
+pub enum JsonLdError {
+    #[error("Could not parse json-ld: `{0}`")]
+    ParseError(String),
+    #[error("Missing @context field")]
+    MissingContext,
+    #[error("Invalid @context: Does not match http://schema.org")]
+    InvalidContext,
+    #[error("Missing @type field")]
+    MissingType,
+    #[error("Invalid @type")]
+    InvalidType,
+}
+
+impl From<serde_json::Error> for JsonLdError {
+    fn from(err: serde_json::Error) -> Self {
+        JsonLdError::ParseError(err.to_string())
+    }
+}
+
+#[inline(never)]
+fn try_parse_author(v: &serde_json::Value) -> Option<String> {
+    match v {
+        serde_json::Value::String(s) => {
+            if s.starts_with("[{") {
```

Comment:
Instead of looking for these characters, could you just try parsing it as JSON?  This'll break if they have their JSON formatted in other valid ways

```
[
   {
      "key": "value",
...
```

etc

Reply:
i don't think they can format it like that because they embed a json blob as a string as a value in a JSON object

Reply:
ugh nevermind you are still right because they can include escpaed `\n` characters.

Reply:
fixed